src/main/java/org/truffleruby/language/yield/YieldExpressionNode.java

```diff
@@ -48,18 +48,18 @@ public YieldExpressionNode(
             ArgumentsDescriptor descriptor,
             RubyNode[] arguments,
             RubyNode readBlockNode,
-            boolean warnInModuleBody) {
+            boolean isModuleBody) {
         super(isSplatted, descriptor);
         this.arguments = arguments;
         this.readBlockNode = readBlockNode;
-        this.warnInModuleBody = warnInModuleBody;
+        this.isModuleBody = isModuleBody;
     }
 
     @ExplodeLoop
     @Override
     public final Object execute(VirtualFrame frame) {
```

Comment:
There are more simplifications to be had with this change.
Notably `InternalMethod.getCapturedBlock()` can be removed entirely.

Reply:
Do you mean that inside the `readBlock` we can drop capturedBlock call entirely and just return `nil`?

I'm not sure that I get the point about dropping `InternalMethod.getCapturedBlock()` because it's used otherwise

Reply:
Yes this whole part is dead, you can replace the whole readBlock() with just `readBlockNode.execute(frame)`

Reply:
Ok, at this point I'm lacking knowledge about the "captured block" thing and can't say - I understand it. I see that we have noCapturedBlock exception earlier and if you say - we don't need to capture block - I'm going to clean it. 

_P.S Still want to know how it became dead or maybe a link to get the context_

Reply:
You can look at all the places the set a non-nil capturedBlock in an InternalMethod, it's only one in ModuleBodyDefinitionNode, and that one doesn't need to set it anymore, it was done specifically for this weird passing of the block through `class << self`. Those semantics are gone now in CRuby thankfully, so we don't need to handle that special case anymore, and so we can remove the field from InternalMethod and simplify every usage of `getCapturedBlock()` to `nil`, which makes this `if (block == nil) { ` fold to false, and the code inside is dead (also remove the now-unused `useCapturedBlock` field).

Reply:
Does that mean that I also have to clean `noCapturedBlock.enter()` call as there will be no capturedBlock things?

Reply:
Yes of course

Reply:
Done